src/goto-harness/goto_harness_parse_options.cpp

```diff
@@ -7,46 +7,168 @@ Author: Diffblue Ltd.
+
+  // Initialise harness generator
+  auto harness_generator =
+    factory.factory(config.harness_type, factory_options);
+  CHECK_RETURN(harness_generator != nullptr);
+
+  // Read goto binary into goto-model
+  auto read_goto_binary_result =
+    read_goto_binary(config.in_file, get_message_handler());
+  if(!read_goto_binary_result.has_value())
+  {
+    throw deserialization_exceptiont{"failed to read goto program from file `" +
+                                     config.in_file + "'"};
+  }
+  auto goto_model = std::move(read_goto_binary_result.value());
+
+  harness_generator->generate(goto_model, config.harness_function_name);
+
+  // Write end result to new goto-binary
+  if(write_goto_binary(config.out_file, goto_model, get_message_handler()))
+  {
+    throw system_exceptiont{"failed to write goto program from file `" +
+                            config.out_file + "'"};
+  }
+
+  return CPROVER_EXIT_SUCCESS;
 }
 
 void goto_harness_parse_optionst::help()
 {
-  std::cout << '\n'
-            << banner_string("Goto-Harness", CBMC_VERSION) << '\n'
-            << align_center_with_border("Copyright (C) 2019") << '\n'
-            << align_center_with_border("Diffblue Ltd.") << '\n'
-            << align_center_with_border("info@diffblue.com")
-            // ^--- No idea if this is the right email address
-            << '\n'
-            << '\n'
-            << "Usage:                       Purpose:\n"
-            << '\n'
-            << " goto-harness [-?] [-h] [--help]  show help\n"
-            << " goto-harness --version           show version\n";
+  status()
+    << '\n'
+    << banner_string("Goto-Harness", CBMC_VERSION) << '\n'
+    << align_center_with_border("Copyright (C) 2019") << '\n'
+    << align_center_with_border("Diffblue Ltd.") << '\n'
+    << align_center_with_border("info@diffblue.com") << '\n'
+    << '\n'
+    << "Usage:                       Purpose:\n"
+    << '\n'
+    << " goto-harness [-?] [-h] [--help]  show help\n"
+    << " goto-harness --version           show version\n"
+    << " goto-harness <in> <out> --harness-function-name <name> --harness-type "
+       "<harness-type> [harness options]\n"
+    << "\n"
+    << "<in> <out>                 goto binaries to read from / write to\n"
+    << "--harness-function-name    the name of the harness function to "
+       "generate\n"
+    << "--harness-type             one of the harness types listed below\n"
+    << "\n\n"
+    << FUNCTION_HARNESS_GENERATOR_HELP << eom;
 }
 
 goto_harness_parse_optionst::goto_harness_parse_optionst(
   int argc,
   const char *argv[])
   : parse_options_baset{GOTO_HARNESS_OPTIONS, argc, argv, ui_message_handler},
+    messaget(ui_message_handler),
     ui_message_handler(cmdline, std::string("GOTO-HARNESS ") + CBMC_VERSION)
 {
 }
+
+goto_harness_parse_optionst::goto_harness_configt
+goto_harness_parse_optionst::handle_common_options()
+{
+  goto_harness_configt config{};
```

Comment:
With the above changes this will shadow the global `config` object, please rename.

Reply:
Please avoid shadowing the global `config` object.